docs/source/example_permanent_magnets.rst

```diff
@@ -0,0 +1,328 @@
+  pol_type = np.concatenate((pol_type, pol_type_f))
+  
+  # Get face-edge orientation vectors for each magnet and reject the negatives 
+  pol_axes_fe_ftri, pol_type_fe_ftri = polarization_axes(['fe_ftri'])
+  ntype_fe_ftri = int(len(pol_type_fe_ftri)/2)
+  pol_axes_fe_ftri = pol_axes_fe_ftri[:ntype_fe_ftri, :]
+  pol_type_fe_ftri = pol_type_fe_ftri[:ntype_fe_ftri] + 1
+  pol_axes = np.concatenate((pol_axes, pol_axes_fe_ftri), axis=0)
+  pol_type = np.concatenate((pol_type, pol_type_fe_ftri))
+  
+  # Get face-corner orientation vectors for each magnet and reject the negatives 
+  pol_axes_fc_ftri, pol_type_fc_ftri = polarization_axes(['fc_ftri'])
+  ntype_fc_ftri = int(len(pol_type_fc_ftri)/2)
+  pol_axes_fc_ftri = pol_axes_fc_ftri[:ntype_fc_ftri, :]
+  pol_type_fc_ftri = pol_type_fc_ftri[:ntype_fc_ftri] + 2
+  pol_axes = np.concatenate((pol_axes, pol_axes_fc_ftri), axis=0)
+  pol_type = np.concatenate((pol_type, pol_type_fc_ftri))
+
+  # Read in the phi coordinates and set the pol_vectors
+  ophi = orientation_phi(fname_corn)[:nMagnets_tot]
+  discretize_polarizations(mag_data, ophi, pol_axes, pol_type)
+  pol_vectors = np.zeros((nMagnets_tot, len(pol_type), 3))
+  pol_vectors[:, :, 0] = mag_data.pol_x
+  pol_vectors[:, :, 1] = mag_data.pol_y
+  pol_vectors[:, :, 2] = mag_data.pol_z
+  
+  # Set pol_vectors during initialization of NCSX PermanentMagnetGrid object
+  kwargs = {"coordinate_flag": "cylindrical", "pol_vectors": pol_vectors}
+  famus_filename = 'init_orient_pm_nonorm_5E4_q4_dp.focus'
+  pm_opt = PermanentMagnetGrid.geo_setup_from_famus(
+      s, Bnormal, famus_filename, **kwargs
+  )
+
+If the PermanentMagnetGrid is initialized with `pol_vectors`, the optimization should
+be performed with the `ArbVec` algorithm variant::
+
+  algorithm = 'ArbVec'
+  R2_history, Bn_history, m_history = GPMO(pm_opt, algorithm, **kwargs)
+
+For best performance, GPMO should be used with backtracking AND orientations for 
+each magnet. Even better, the backtracking can be improved by specifying `thresh_angle`,
+which determines the minimum angle between two nearby magnets to consider removing them
+during backtracking (the default backtracking algorithm assumes that thresh_angle = 180,
+i.e. fully oppositely oriented, but when there are many orientations available to the magnets,
+often the magnets are not fully opposite). Putting this altogether, the most effective
+use of the GPMO algorithm and its variants looks something like::
+
+  algorithm = 'ArbVec_backtracking'
+  kwargs = initialize_default_kwargs('GPMO')
+  kwargs['K'] = 20000 
+  kwargs['dipole_grid_xyz'] = np.ascontiguousarray(pm_opt.dipole_grid_xyz)
+  if algorithm == 'backtracking' or algorithm == 'ArbVec_backtracking':
+      kwargs['backtracking'] = 200 
+      kwargs['Nadjacent'] = 10  
+      kwargs['dipole_grid_xyz'] = np.ascontiguousarray(pm_opt.dipole_grid_xyz)  # make sure C++ compatible
+      if algorithm == 'ArbVec_backtracking':
+          kwargs['thresh_angle'] = 120
```

Comment:
In the MUSE and PM4STELL examples you use radians.

Reply:
fixed, thanks